source/main.cpp

```diff
@@ -258,6 +262,16 @@ void GameLoop(PlayerInfo &player, Conversation &conversation, bool &debugMode)
 		// Tell all the panels to step forward, then draw them.
 		((!isPaused && menuPanels.IsEmpty()) ? gamePanels : menuPanels).StepAll();
 		
+		// Currently running the hardcoded "empty" testcase:
+		//   Wait for the game to be fully loaded and then quit.
+		//
+		// This testcase can catch issues related to startup/data-loading,
+		// and it helps to show if the CI testframework is working.
+		// This hardcoded testcase is expected to be replaced by the
+		// larger testframework that is planned for ES.
+		if(!testToRun.empty() && GameData::IsLoaded())
+			menuPanels.Quit();
+		
 		// Caps lock slows the frame rate in debug mode.
 		// Slowing eases in and out over a couple of frames.
 		if((mod & KMOD_CAPS) && inFlight && debugMode)
```

Comment:
The idea of the current location is that the test-code is safely placed after all processing for the step (events, local logic, stepping of the panels) is done and the game is in a consistent state.

MenuPanels.StepAll() performs processing like pushing or popping menu-panels from the UI-stack, so if we run the testcode before the MenuPanels.StepAll, then it is possible that some actions got triggered that will add or remove menus on the stack, but that the menus themselves have not been pushed or popped yet. Having the test-code interact with the menus just before MenuPanels.StepAll() sounds risky to me, especially since the game itself does not support giving inputs at that moment.

I prefer not to place the test-code after DrawAll, because skipFrame (on line 281 before this change, line 295 in this change) causes DrawAll to be called less often when fast-forward is active. Calling the test-functions with variable timing can make it more difficult to use timeout-counters in the test.

Another location that I would consider consistent and safe for calling the test-code would be the beginning of this loop (before the `while(SDL_PollEvent)`).